tools/clang/lib/Basic/DiagnosticIDs.cpp

```diff
@@ -540,7 +545,10 @@ static bool getDiagnosticsInGroup(diag::Flavor Flavor,
   // Add the members of the option diagnostic set.
   const int16_t *Member = DiagArrays + Group->Members;
   for (; *Member != -1; ++Member) {
-    if (GetDiagInfo(*Member)->getFlavor() == Flavor) {
+    // HLSL Change: Check result of GetDiagInfo
+    const StaticDiagInfoRec *Info = GetDiagInfo(*Member);
+    assert(Info && "otherwise, group contains invalid diag ID");
```

Comment:
Should we be using `DXASSERT` here and below?

Reply:
This is not in HLSL-specific code, so I think it's more appropriate to use assert here.